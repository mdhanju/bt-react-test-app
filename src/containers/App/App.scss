/* Container styles import here */

#bt-web-appname {
    display: flex;
    flex-direction: column;
    height: 100%;

    .container-fluid {
        padding-right: 3rem;
        padding-left: 3rem;

        @include media ($device) {
            padding-right: 1.6rem;
            padding-left: 1.6rem;
        }
    }

    &>* {
        display: flex;
    }

    &>main {
        flex-grow: 1;
        padding: 1.4rem 0;
        flex-direction: column;

        @include media-breakpoint-down(sm) {
            &.container-fluid {
                padding: 0;
            }
        }

        &>* {
            display: flex;
        }
    }

    .cp-main-section {
        flex-grow: 1;
        justify-content: center;

        @include media-breakpoint-down(sm) {
            padding-left: 1.6rem;
            padding-right: 1.6rem;
            padding-bottom: 2rem;
        }
    }

    .under-construction {
        display: flex;
        align-items: center;
        font-size: 2rem;
    }
}

.mobile-hide {
    @include media-breakpoint-down(sm) {
        display: none;
    }
}

.desktop-hide {
    @include media-breakpoint-up(md) {
        display: none;
    }
}
