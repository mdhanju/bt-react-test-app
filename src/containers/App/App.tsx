import React from 'react'
import { connect } from 'react-redux'
import { initializeApp } from '../../store/actions/applicationInitActions'
import { ThunkDispatch } from 'redux-thunk'
import AppRouter from '../../components/AppRouter'

import { State } from '../../store/reducers/rootReducer'
import { ApplicationInitAction } from '../../store/actions/applicationInitActions'

type Props = {
    isAppInitialized: string;
    initializeApp: () => void;
};

class App extends React.Component<Props> {
    componentDidMount(): void {
        this.props.initializeApp()
    }

    render(): React.ReactNode {
        return <AppRouter isAppInitialized={this.props.isAppInitialized} />
    }
}

const mapStateToProps = (state: State): object => ({
    isAppInitialized: state.appContext.isAppInitialized
})

const mapDispatchToProps = (
    dispatch: ThunkDispatch<State, void, ApplicationInitAction>
): object => ({
    initializeApp: (): Promise<void> => dispatch(initializeApp())
})

const AppWithRedux: /* eslint-disable-line @typescript-eslint/no-explicit-any */ any = connect(
    mapStateToProps,
    mapDispatchToProps
)(App)
export default AppWithRedux
