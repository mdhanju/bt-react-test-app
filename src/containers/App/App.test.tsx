/* eslint-disable react/jsx-no-undef */

import React from 'react'
import App from '../App'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])

describe('<App />', () => {
    it('should render initial layout', () => {
        const FakeInit = jest.fn()
        const wrapper = shallow(
            <App isAppInitialized="INPROGRESS" initializeApp={FakeInit} />
        )
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
        expect(FakeInit).toHaveBeenCalled()
    })

    it('should render Div with Text', () => {
        const wrapper = render(
            <Intl locale="en">
                <App isAppInitialized="INPROGRESS" initializeApp={jest.fn()} />
            </Intl>
        )
        expect(wrapper.text()).toBe(
            `Sample AppÂ© ${new Date().getFullYear()}\u00a0Blue Triangle, Inc.`
        )
        expect(wrapper.html()).toContain('div')
    })

    xit('should mount Div with Text', () => {
        const initialState = {
            appContext: {
                isAppInitialized: 'INPROGRESS',
                appConfig: {}
            }
        }
        const store = mockStore(initialState)
        //const mockInitfn = jest.fn();
        store.dispatch = jest.fn()
        const wrapper = mount(
            <Intl locale="en">
                <App store={store} initializeApp={mockInitfn} />
            </Intl>
        )

        expect(mockInitfn.mock.calls.length).toBe(1)
    })
})
