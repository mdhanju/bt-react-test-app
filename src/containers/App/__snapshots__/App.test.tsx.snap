// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render initial layout 1`] = `
Array [
  <AppRouter
    isAppInitialized="INPROGRESS"
  />,
]
`;
